canister/src/validation.rs

```diff
@@ -0,0 +1,57 @@
+use crate::{blocktree::BlockDoesNotExtendTree, state::State, unstable_blocks, Block};
+use bitcoin::BlockHeader;
+use ic_btc_validation::HeaderStore;
+
+/// A structure passed to the validation crate to validate a specific block header.
+pub struct ValidationContext<'a> {
+    state: &'a State,
+    chain: Vec<&'a Block>,
+}
+
+impl<'a> ValidationContext<'a> {
```

Comment:
Note that there are no unit tests in this file, and that's because the current test suite already covers this code very well, so I didn't see a need to add additional tests, but I can add them if requested.

Reply:
I think ideally we'd have some tests that cover this module in isolation as it's used as a building block for the validation but I don't feel so strongly to block the PR on it.

Reply:
Done.